### PR TITLE
Migrate Learning Modules to have exactly one Plc::CourseUnit

### PR DESCRIPTION
**Pull request by joshlory**
The Courses tree view now includes Learning Modules under Course Units:

screen shot 2016-04-07 at 4 35 08 pm

### CODE REVIEW

File: dashboard/app/models/plc/learning_module.rb [hunk 1]
```diff
@@ -2,14 +2,22 @@
...
+#  index_plc_learning_modules_on_plc_course_unit_id  (plc_course_unit_id)
 #
 
 class Plc::LearningModule < ActiveRecord::Base
+  belongs_to :plc_course_unit, class_name: '::Plc::CourseUnit', foreign_key: 'plc_course_unit_id'
   has_and_belongs_to_many :plc_tasks, class_name: '::Plc::Task', foreign_key: 'plc_learning_module_id', association_foreign_key: 'plc_task_id'
   has_many :plc_module_assignments, class_name: '::Plc::EnrollmentModuleAssignment', foreign_key: 'plc_learning_module_id', dependent: :destroy
   has_many :plc_evaluation_answers, class_name: '::Plc::EvaluationAnswer', foreign_key: 'plc_learning_module_id', dependent: :destroy
+
+  validates_presence_of :plc_course_unit_id
```
Comment (joshlory) [inline dashboard/app/models/plc/learning_module.rb:22]: I should make this `readonly` as well.

File: dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class AddCourseUnitToLearningModules < ActiveRecord::Migration
+  def up
+    add_reference :plc_learning_modules, :plc_course_unit, index: true, foreign_key: true
+
+    # Create a default Plc::CourseUnit to own all orphaned Learning Modules
+    default_course = Plc::Course.find_or_create_by! name: 'Default Course'
```
Comment (mehalshah) [inline dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb:6]: Course doesn't need any params to exist, and if for some dumb reason there's a 'default course' then you'll be messing with it. You can just do Plc::Course.create(). For course unit, you'll have to specify the default_course but nothing else.

Alternatively - give the course a name indicating that it exists for orphaned modules on X date

File: dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    Plc::CourseUnit.all.each do |unit|
+      unit.plc_evaluation_questions.each do |question|
+        question.plc_evaluation_answers.each do |answer|
+          answer.plc_learning_module.update! plc_course_unit_id: unit.id
+        end
+      end
+    end
+
+    change_column_null :plc_learning_modules, :plc_course_unit_id, false
+  end
```
Comment (mehalshah) [inline dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb:25]: If the default course and course unit have no modules, then delete them.

**Comment by mehalshah** [comment]
Logic LGTM, some concerns around the migration.

File: dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb (hunk 2 above)
Comment (joshlory) [inline dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb:6]: I'd rather add it to a course with a specific name so running the migration forwards and backwards a number of times doesn't create a bunch of new Courses.

File: dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb (hunk 3 above)
Comment (joshlory) [inline dashboard/db/migrate/20160407173756_add_course_unit_to_learning_modules.rb:25]: Good call, fixed in 18027f9c2c293c36594f6afc5f7ff970a95010e0.

**Comment by mehalshah** [comment]
Looks good

**Pull request merged**